File: src/main/java/kitchenpos/eatinorders/tobe/domain/EatInOrderRepository.java
package kitchenpos.eatinorders.tobe.domain;

import kitchenpos.eatinorders.tobe.domain.constant.EatInOrderStatus;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface EatInOrderRepository {
    EatInOrder save(EatInOrder order);

    Optional<EatInOrder> findById(UUID id);

    List<EatInOrder> findAll();

    boolean existsByOrderTableAndStatusNot(TobeOrderTable orderTable, EatInOrderStatus eatInOrderStatus);
}

File: src/main/java/kitchenpos/eatinorders/tobe/domain/EatInOrderRepository.kt
package kitchenpos.eatinorders.tobe.domain

import java.util.*

fun EatInOrderRepository.findByIdOrNull(eatInOrderId: UUID) = findById(eatInOrderId).orElse(null)

interface EatInOrderRepository {

    fun save(eatInOrder: EatInOrder): EatInOrder

    fun existsByOrderTableIdAndStatusNot(orderTableId: UUID, status: EatInOrderStatus): Boolean

    fun findById(eatInOrderId: UUID): Optional<EatInOrder>

}
